audformat/core/utils.py

```diff
@@ -583,17 +589,14 @@ def intersect(
 
         # if we have a mixture
         # of pd.Index and pd.MultiIndex
-        # convert all to pd.MultiIndex
+        # convert all to pd.Index
         if (
             objs[0].nlevels == 1
             and len(set(isinstance(obj, pd.MultiIndex) for obj in objs)) == 2
         ):
             objs = [
-                obj if isinstance(obj, pd.MultiIndex)
-                else pd.MultiIndex.from_arrays(
-                    [obj.to_list()],
-                    names=[obj.name],
-                )
+                obj if not isinstance(obj, pd.MultiIndex)
+                else obj.get_level_values(0)
                 for obj in objs
             ]
```

Comment:
In concat we do

```python
for n, obj in enumerate(objs):
    if isinstance(obj.index, pd.MultiIndex):
        objs[n].index = obj.index.get_level_values(0)
```
here we do
```python
objs = [
    obj if not isinstance(obj, pd.MultiIndex)
    else obj.get_level_values(0)
    for obj in objs
]
```
I think readability would be improved if we decide for only one solutuion.

We also need it again in `union()`, so maybe add a helper function that includes the `if`-statement as well.

Reply:
There is a difference between the two cases - in the first case we iterate over a list of `pd.Series` or `pd.DataFrame`, in the second one over a list with index objects.

Reply:
But adding a utility function makes of course sense.

Reply:
I added a helper function `_convert_single_level_multi_index()` that is now used in all three places. I also renamed `same_dtype()` to `_is_same_dtype()` as this function is also not part of the API.

Reply:
Cool, that make it indeed much nicer inside `concat()`, `intersect()`, `union()`